dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/analytics/generator/GeneratorHelper.java

```diff
@@ -0,0 +1,343 @@
+         */
+        public class _className extends AnalyticsApiTest {
+          _actionInstance
+        """
+        .replace("_topComment", TOP_CLASS_COMMENT)
+        .replace("_className", className)
+        .replace("_actionInstance", ACTION_INSTANCE);
+  }
+
+  /**
+   * Generates the test method for the given method "name" and "url".
+   *
+   * @param name the method's name.
+   * @param url the full URL associated with this test.
+   * @return the full test method.
+   */
+  static String generateTestMethod(String name, String url) {
+    if (isBlank(name) || isBlank(url)) {
+      return EMPTY;
+    }
+
+    String testDeclaration = "@Test\n public void " + name + "() throws JSONException {\n";
+
+    StringBuilder params = buildTestParams(url);
+    String dimensionUid = getTargetDim(url);
+
+    String testTarget = "\n// When \nApiResponse response = actions.get(params);\n";
+
+    // If a target dimension is present, it should be used as part of the action call.
+    if (isNotBlank(dimensionUid)) {
+      testTarget =
+          "\n// When\n ApiResponse response = actions."
+              + ACTION
+              + "().get(\""
+              + dimensionUid
+              + "\", JSON, JSON, params);\n";
+    }
+
+    String response = getResponse(url);
+
+    if (isNotBlank(response)) {
+      ReadContext ctx = JsonPath.parse(getResponse(url));
+
+      String headersAssertion = buildHeadersAssertion(ctx);
+      String responseAssertions = buildResponseAssertion(ctx);
+      String metaDataAssertion = buildMetaDataAssertion(ctx);
+      String rowsAssertion = buildRowsAssertion(ctx);
+
+      return testDeclaration
+          + params
+          + testTarget
+          + responseAssertions
+          + metaDataAssertion
+          + headersAssertion
+          + rowsAssertion
+          + "}";
+    } else {
+      System.err.println("## No Response. Check the URL requested.");
+      return EMPTY;
+    }
+  }
+
+  /**
+   * Builds the assertions for all rows.
+   *
+   * @param ctx the {@link ReadContext}. The values for the assertions are extracted from it.
+   * @return the built assertions.
+   */
+  private static String buildRowsAssertion(ReadContext ctx) {
+    List<Map<String, Object>> rows = ctx.read("$.rows");
+    StringBuilder rowsAssertion = new StringBuilder("\n// Assert rows.\n");
+
+    for (int i = 0; i < rows.size(); i++) {
+      JSONArray jsonArray = (JSONArray) rows.get(i);
+
+      if (ASSERT_ROW_INDEX) {
+        rowsAssertion.append("validateRow(response," + i + ", List.of(");
+      } else {
+        rowsAssertion.append("validateRow(response, List.of(");
+      }
+
+      rowsAssertion
+          .append(
+              jsonArray.stream()
+                  .map(
+                      v ->
+                          "\""
+                              + v.toString().replace(System.getProperty("line.separator"), "\\n")
+                              + "\"")
+                  .collect(joining(",")))
```

Comment:
Could this possibly be ?
```suggestion
                      .map(Object::toString)
                      .map(v -> v.replace(System.getProperty("line.separator"), "\\n"))
                      .collect(joining(",", "\"", "\"")))
```

Reply:
Done